broker/src/test/java/org/apache/rocketmq/broker/processor/SendMessageProcessorTest.java

```diff
@@ -226,4 +238,75 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                return subscriptionGroupManager;
+            }
+        }).when(brokerController).getSubscriptionGroupManager();
+        final StringBuilder resultCollector = new StringBuilder();
+
+        MessageExt messageExt = new MessageExt();
+        messageExt.setTopic(topic);
+        messageExt.setQueueId(0);
+        messageExt.setBody("simple message".getBytes());
+        messageExt.setCommitLogOffset(876867867L);
+        messageExt.setReconsumeTimes(16);
+        messageExt.setMsgId(msgId);
+        messageExt.setStoreTimestamp(System.currentTimeMillis());
+        messageExt.putUserProperty("test222","gggg");
+        messageExt.getProperties().put(MessageConst.PROPERTY_UNIQ_CLIENT_MESSAGE_ID_KEYIDX,msgId);
+        messageExt.getProperties().put(MessageConst.PROPERTY_RETRY_TOPIC,topic);
+        when(messageStore.lookMessageByOffset(anyLong())).thenReturn(messageExt);
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                Object[] arguments = invocationOnMock.getArguments();
+                resultCollector.append(arguments[0]);
+                return null;
+            }
+        }).when(dlqLogger).info(anyString());
+
+        Field dlqLoggerField = SendMessageProcessor.class.getDeclaredField("dlqLogger");
+        dlqLoggerField.setAccessible(true);
+        dlqLoggerField.set(sendMessageProcessor,dlqLogger);
+
+        when(messageStore.putMessage(any(MessageExtBrokerInner.class))).thenReturn(new PutMessageResult(PutMessageStatus.PUT_OK, new AppendMessageResult(AppendMessageStatus.PUT_OK)));
+
+        ConsumerSendMsgBackRequestHeader consumerSendMsgBackRequestHeader = new ConsumerSendMsgBackRequestHeader();
+        consumerSendMsgBackRequestHeader.setGroup(groupName);
+        consumerSendMsgBackRequestHeader.setMaxReconsumeTimes(3);
+        consumerSendMsgBackRequestHeader.setOffset(876867867L);
+        consumerSendMsgBackRequestHeader.setDelayLevel(10);
+        consumerSendMsgBackRequestHeader.setOriginTopic("%RETRY%"+groupName);
+        consumerSendMsgBackRequestHeader.setOriginMsgId(msgId);
+
+        RemotingCommand requestCommand = RemotingCommand.createRequestCommand(RequestCode.CONSUMER_SEND_MSG_BACK, consumerSendMsgBackRequestHeader);
+        requestCommand.addExtField("group",groupName);
+        requestCommand.addExtField("maxReconsumeTimes","3");
+        requestCommand.addExtField("offset","876867867");
+        requestCommand.addExtField("delayLevel","10");
+        requestCommand.addExtField("originTopic","%RETRY%"+groupName);
+        requestCommand.addExtField("originMsgId",msgId);
+        sendMessageProcessor.processRequest(null,requestCommand);
+        String result = resultCollector.toString();
+        Assert.assertTrue(result.contains(msgId));
+        Assert.assertTrue(result.contains(topic));
```

Comment:
Can we use the unified `assertThat` of `assertj`?

Reply:
Yeah, assertj has been fixed to assertThat.